google_drive/google_calendar.py

```diff
@@ -35,62 +37,85 @@ class GoogleCalendarBlock(XBlock):
         display_name="Display Name",
         help="This name appears in the horizontal navigation at the top of the page.",
         scope=Scope.settings,
-        default="Example Google Calendar"
+        default="Google Calendar"
+    )
+
+    calendar_id = String(
+        display_name="Public Calendar ID",
+        help=(
+            "Google provides an ID for publicly available calendars. In the Google Calendar, "
+            "open Settings and copy the ID from the Calendar Address section into this field."
+        ),
+        scope=Scope.settings,
+        default=DEFAULT_CALENDAR_ID
     )
 
-    embed_code = String(
-        display_name="Embed Code",
-        help="Google provides an embed code for Drive calendars with a variety of settings. From inside a Google Drive calendar, select Publish to the Web from within the File menu to get to your Embed Code with your desired settings.",
+    default_view = Integer(
+        display_name="Default View",
+        help="The calendar view that students see by default. A student can change this view.",
         scope=Scope.settings,
-        default=textwrap.dedent("""
-            <iframe
-                src="https://docs.google.com/presentation/d/1B22AsMwG7jgE39vm1BuiJClir_JUz1q077cPEKJR2mI/embed?start=false&loop=false&delayms=3000"
-                frameborder="0"
-                width="960"
-                height="569"
-                allowfullscreen="true"
-                mozallowfullscreen="true"
-                webkitallowfullscreen="true">
-            </iframe>
-        """))
-
-    def student_view(self, context={}):
+        default=1
+    )
+
+    views = [(0, 'Week'), (1, 'Month'), (2, 'Agenda')]
+
+    # Context argument is specified for xblocks, but we are not using herein
+    def student_view(self, context):  # pylint: disable=unused-argument
         """
         Player view, displayed to the student
         """
-
         fragment = Fragment()
-        context.update({
+
+        view = self.views[self.default_view][1]
+        iframe = CALENDAR_IFRAME.format(view, self.calendar_id, self.display_name)
```

Comment:
Actually, now that I read more code, why is CALENDAR_IFRAME a separate format string? Wouldn't it be better to just include that HTML into google_calendar.html?  It's only used in this one place.